metadatastore/odm_templates.py

```diff
@@ -23,7 +23,7 @@ class BeamlineConfig(DynamicDocument):
     config_params = DictField(required=False, unique=False)
     uid = StringField(required=True, unique=True)
     time = FloatField(required=True)
-    meta = {'indexes': ['-_id'], 'db_alias': ALIAS}
+    meta = {'indexes': ['-_id', '-uid'], 'db_alias': ALIAS}
```

Comment:
Because the default indexing of mongoengine is ascending order. We want to ensure _id are done in descending order for quicker search. _id is the primary key for retrieving the actual documents in the backend. Specifying that “-“ makes the queries a little bit faster with growing document size.


On May 26, 2015, at 9:24 AM, Eric Dill <notifications@github.com<mailto:notifications@github.com>> wrote:


Then I don't understand why id is indexed on all of the documents